### ISSUE TITLE
StdString.h spawning tons of warning C4396 on Visual Studio 2015.

### ISSUE THREAD

**Issue by Prcuvu** [stepmania/stepmania#784]
When compiling StepMania on Visual Studio 2015, the output dialog was flooded with C4396 warnings as shown below:
```
StdString.h(630): warning C4396: 'StdString::operator +': the inline specifier cannot be used when a friend declaration refers to a specialization of a function template
StdString.h(631): warning C4396: 'StdString::operator +': the inline specifier cannot be used when a friend declaration refers to a specialization of a function template
StdString.h(632): warning C4396: 'StdString::operator +': the inline specifier cannot be used when a friend declaration refers to a specialization of a function template
StdString.h(633): warning C4396: 'StdString::operator +': the inline specifier cannot be used when a friend declaration refers to a specialization of a function template
```
I would try using the latest version of `StdString.h` to see if this problem can be avoided.

**Comment by wolfman2000** [comment]
One slight problem with that. Do you know of any latest version? Technically, that would be ours, and it was modified partially to deal with internal exclusive stuff.

What I am hoping to do is replace `StdString.h` with a plain c++ `string`. It's not as simple as replacing the `typedef` of `StdString::CStdString`, however. I have run into an issue where trying to do that results in breaking the virtual file system. There is something else going on that needs to be resolved first.

**Comment by Prcuvu** [comment]
The latest version has been put on [CodeProject] (http://www.codeproject.com/Articles/1146/CString-clone-Using-Standard-C). I had a brief test with it and found identifiers colliding with StepMania.
For instance, the new `StdString.h` includes `comdef.h`, which defines a struct `Font`, that collides with a class of the same name. Macro `TRACE` also collides with one defined in `global.h`.
Maybe some redundant code can be removed, or it is not even a good idea to upgrade this header.

**Comment by wolfman2000** [comment]
So there are updates to the base code. Either that or it's the same code as before but uploaded on a different site. Either way, the changes that were made to it directly by prior StepMania developers won't be there.

I will not be personally updating `StdString.h` if I can help it. However, if you wish to try to make that work in a PR, I will consider merging it.

**Comment by Prcuvu** [comment]
PR #833 kills the warnings, but there are still tons of problems to deal with.

**Comment by Prcuvu** [comment]
Closed by #952.
